deployment/postgres-init-db/sql/tables/scheduler/scheduling_goal_definition.sql

```diff
@@ -13,19 +16,39 @@ create table scheduler.scheduling_goal_definition(
     references scheduler.scheduling_goal_metadata
     on update cascade
     on delete cascade,
+  constraint scheduling_procedure_has_uploaded_jar
+    foreign key (uploaded_jar_id)
+      references merlin.uploaded_file
+      on update cascade
+      on delete restrict,
   constraint goal_definition_author_exists
     foreign key (author)
     references permissions.users
     on update cascade
-    on delete set null
+    on delete set null,
+  constraint check_goal_definition_type_consistency
+    check (
+      (type = 'EDSL' and definition is not null and uploaded_jar_id is null)
+        or
+      (type = 'JAR' and uploaded_jar_id is not null and definition is null)
+    )
```

Comment:
This check constraint seems sufficiently complex enough for a small DB test.

Reply:
Pushed a small DB test under `SchedulingDatabaseTests.ProceduralSchedulingTests`

Reply:
Done! Added a few more test cases